NossoMercadoLivre.API/Controllers/UserController.cs

```diff
@@ -17,18 +18,22 @@ public class UserController : ControllerBase
     {
 
         [HttpPost]
-        public ActionResult<User> NewUser([FromServices] IUserService userService, [FromBody] UserDTO user)
+        public ActionResult<User> NewUser([FromServices] IUserRepository userRepository, [FromBody] UserDTO userDto)
         {
-            try
+            UserDTOValidator validator = new UserDTOValidator();
+            var validateResult = validator.Validate(userDto);
+            if (validateResult.IsValid)
```

Comment:
*A complicação do nosso é código é proporcional a complicação da nossa feature. Quanto mais simples, melhor.*
+
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Opa, por que você prefere trazer uma classe com configuração de validação e aplicar a validação na mão quando tudo isso já está pronto no framework escolhido?

Reply:
Pois é mais agradável quando se trata de uma validação customizada, como por exemplo um CPF. No momento não tem, mas considerando o problema com certeza irá ter dado o problema.

Reply:
Oi Erick, entendo. Uma régua do nosso treinamento é que ser agradável não conta para decidirmos qual código vamos escrever. Fazemos código com dificuldade bem distribuída pelo sistema e só elevamos essa dificuldade quando é necessário. Recomendo usar o que já vem pronto no framework e desencanar dessa customização. Inclusive o CPF pode ser criado pelo mecanismo de validação pronta do asp.net core.